libsolidity/analysis/DocStringAnalyser.cpp

```diff
@@ -127,133 +111,41 @@ bool DocStringAnalyser::visit(EventDefinition const& _event)
-	checkParameters(_callable, _node, _annotation);
-
-	if (
+	if (CallableDeclaration const* baseFunction = resolveInheritDoc(_callable.annotation().baseFunctions, _node, _annotation))
+		copyMissingTags(_annotation, {baseFunction});
+	else if (
 		_annotation.docTags.empty() &&
 		_callable.annotation().baseFunctions.size() == 1 &&
 		parameterNamesEqual(_callable, **_callable.annotation().baseFunctions.begin())
 	)
 		copyMissingTags(_annotation, _callable.annotation().baseFunctions);
-
-	if (_node.documentation() && _annotation.docTags.count("author") > 0)
-		m_errorReporter.warning(
-			9843_error, _node.documentation()->location(),
-			"Documentation tag @author is only allowed on contract definitions. "
-			"It will be disallowed in 0.7.0."
-		);
 }
 
-void DocStringAnalyser::parseDocStrings(
+CallableDeclaration const* DocStringAnalyser::resolveInheritDoc(
+	std::set<CallableDeclaration const*>& _baseFuncs,
 	StructurallyDocumented const& _node,
-	StructurallyDocumentedAnnotation& _annotation,
-	set<string> const& _validTags,
-	string const& _nodeName
+	StructurallyDocumentedAnnotation& _annotation
 )
 {
-	DocStringParser parser;
-	if (_node.documentation() && !_node.documentation()->text()->empty())
-	{
-		parser.parse(*_node.documentation()->text(), m_errorReporter);
-		_annotation.docTags = parser.tags();
-	}
+	if (_annotation.inheritdocReference == nullptr)
+		return nullptr;
+
+	if (auto const callable = findBaseCallable(_baseFuncs, _annotation.inheritdocReference->id()))
+		return callable;
 
-	size_t returnTagsVisited = 0;
-	for (auto const& docTag: _annotation.docTags)
-	{
-		if (!_validTags.count(docTag.first))
-			m_errorReporter.docstringParsingError(
-				6546_error,
-				_node.documentation()->location(),
-				"Documentation tag @" + docTag.first + " not valid for " + _nodeName + "."
-			);
-		else
-			if (docTag.first == "return")
-			{
-				returnTagsVisited++;
-				if (auto* varDecl = dynamic_cast<VariableDeclaration const*>(&_node))
-				{
-					if (!varDecl->isPublic())
-						m_errorReporter.docstringParsingError(
-							9440_error,
-							_node.documentation()->location(),
-							"Documentation tag \"@" + docTag.first + "\" is only allowed on public state-variables."
-						);
-					if (returnTagsVisited > 1)
-						m_errorReporter.docstringParsingError(
-							5256_error,
-							_node.documentation()->location(),
-							"Documentation tag \"@" + docTag.first + "\" is only allowed once on state-variables."
-						);
-				}
-				else if (auto* function = dynamic_cast<FunctionDefinition const*>(&_node))
-				{
-					string content = docTag.second.content;
-					string firstWord = content.substr(0, content.find_first_of(" \t"));
+	m_errorReporter.docstringParsingError(
+		4682_error,
+		_node.documentation()->location(),
+		"Documentation tag @inheritdoc references contract \"" +
+		_annotation.inheritdocReference->name() +
+		"\", but the contract does not override this function."
```

Comment:
shouldn't it be
"...contract does not contain a function that is overridden by this function"?